clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -2923,14 +2923,20 @@ static void handleWorkGroupSize(Sema &S, Decl *D, const ParsedAttr &AL) {
     return;
 
   uint32_t WGSize[3];
+  if (AL.getNormalizedFullName() == "intel::reqd_work_group_size") {
```

Comment:
getAttributeSpellingListIndex exists for this purpose.

Reply:
But in this situation this statement sifts not just ReqdWorkGroupSize spellings, but also different WorkGroupAttr.

Reply:
I think, that getNormalizedFullName() more applicable here.

Reply:
We have 2 attributes with the same spelling!?!  That is completely unacceptable.

Reply:
No, but getAttributeSpellingListIndex() returns unsigned and an attribute spelling index is a just unsigned in fact. And in result we get the same outcome (true) with diff spellings indices of diff attributes.

Reply:
Then this should check your attribute type as well.  getNormalizedFullName isn't a stable interface.

Reply:
Okey.

Reply:
Ah... I think I see the problem in TableGen.  I'm going to push a fix to LLOrg in the next hour or so to make it so that this enum still gets emitted, since it is useful here.

Reply:
Suggested a review, but my change is here: https://reviews.llvm.org/D76289

Reply:
There, got it in!  rG661c950630fb.  Once that comes down, this line should be AL.getKind == ?? && AL.getSpelling == ReqdWorkGroupSize::CXX11_intel_reqd_work_group_size.

Reply:
Should I wait until the merge?